lib/logdna.rb

```diff
@@ -82,78 +59,66 @@ def level=(value)
       @level = value
     end
 
-    def log(msg=nil, opts={})
-      loggerExist?
-      message = msg
-      message = yield if msg.nil? && block_given?
-      @response = @@client.buffer(message, default_opts.merge(opts).merge({
-            timestamp: (Time.now.to_f * 1000).to_i
-        }))
-      'Saved'
+    def log(message = nil, opts = {})
+      return if message.nil? || message.empty?
+
+      message = message.to_s.encode("UTF-8")
+      @client.write_to_buffer(message, default_opts.merge(opts).merge(
+                                         timestamp: (Time.now.to_f * 1000).to_i
+                                       ))
     end
 
     Resources::LOG_LEVELS.each do |lvl|
       name = lvl.downcase
 
-      define_method name do |msg=nil, opts={}, &block|
-        self.log(msg, opts.merge({
-          level: lvl,
-        }), &block)
+      define_method name do |msg = nil, opts = {}|
```

Comment:
So this is the part I wanted you to understand. What this is (or was) doing is defining methods for each log level `error', `warn`, `debug`, etc. And in doing so, it's mimicking the methods that the Ruby Logger has: https://ruby-doc.org/stdlib-2.4.0/libdoc/logger/rdoc/Logger.html#method-i-info. If you look at all of those methods (I linked to `info`, but it's there for all of them), you'll see that they all take a `&block` that `Evaluates to the message to log.`. You're removing the functionality by removing the block handling in `log`. If you want to do that we can, but note that it does not deviate from what Ruby's Logger looks like.

Reply:
The original code does not replicate the Ruby Logger behavior. I'll make a change so it will d replicate it

Reply:
Actually, it makes sense why the original code does not fully replicate the Ruby Logger behavior. I have restored it.